underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java

```diff
@@ -351,17 +351,31 @@ public boolean mkdirs(String path, MkdirsOptions options) throws IOException {
         // Create directories one by one with explicit permissions to ensure no umask is applied,
         // using mkdirs will apply the permission only to the last directory
         Stack<Path> dirsToMake = new Stack<>();
+        Stack<String> dirsToSetOwner = new Stack<>();
         dirsToMake.push(hdfsPath);
         Path parent = hdfsPath.getParent();
         while (!mFileSystem.exists(parent)) {
           dirsToMake.push(parent);
           parent = parent.getParent();
         }
         while (!dirsToMake.empty()) {
-          if (!FileSystem.mkdirs(mFileSystem, dirsToMake.pop(),
+          Path dirToMake = dirsToMake.pop();
+          if (!FileSystem.mkdirs(mFileSystem, dirToMake,
               new FsPermission(options.getPermission().getMode().toShort()))) {
             return false;
           }
+          dirsToSetOwner.push(dirToMake.toString());
```

Comment:
Could we set the permission as the directories are created? In the case creating one of the directories fails, we would get some leftover directories owned by the wrong owner.